src/main/java/com/google/cloud/tools/appengine/AppEngineDescriptor.java

```diff
@@ -100,11 +100,11 @@ public String getServiceId() {
   }
 
   /**
-   * @return true if the runtime specified by the user is Java8.
+   * @return true if the runtime read from appengine-web.xml is Java8
    */
   public boolean isJava8() {
     String runtime = getRuntime();
-    return runtime != null && runtime.startsWith("java8");
+    return "java8".equals(runtime) || "java8g".equals(runtime);
```

Comment:
There are no other currently valid runtimes that use appengine-web.xml (instead of app.yaml) though. And so far we don't seem to need that.